crates/iceberg/src/spec/values.rs

```diff
@@ -65,24 +65,14 @@ pub enum PrimitiveLiteral {
     Float(OrderedFloat<f32>),
     /// Stored as 8-byte little-endian
     Double(OrderedFloat<f64>),
-    /// Stores days from the 1970-01-01 in an 4-byte little-endian int
-    Date(i32),
-    /// Stores microseconds from midnight in an 8-byte little-endian long
-    Time(i64),
-    /// Timestamp without timezone
-    Timestamp(i64),
-    /// Timestamp with timezone
-    Timestamptz(i64),
     /// UTF-8 bytes (without length)
     String(String),
-    /// 16-byte big-endian value
-    Uuid(Uuid),
-    /// Binary value
-    Fixed(Vec<u8>),
     /// Binary value (without length)
     Binary(Vec<u8>),
     /// Stores unscaled value as big int. According to iceberg spec, the precision must less than 38(`MAX_DECIMAL_PRECISION`) , so i128 is suit here.
     Decimal(i128),
+    /// 16-byte big-endian value
+    Uuid(Uuid),
```

Comment:
I think this is unlike `Timestamp(i64)`. The benefit of remaining this is that we can reuse some implementation of `Uuid`, e.g. when format.🤔

Reply:
But we use `Binary(Vec<u8>)` now, I think for now it will introduce copy overhead when we convert to uuid.🤔 Using `Bytes` can avoid this because Bytes can share the inner buffer.

Reply:
I find that we need to check the length when we use Binary(Vec<u8>) to represent Uuid. How about use i128 to store the Uuid.🤔

Reply:
Uuid needs u128.😭 One option is to introduce U128(u128) now or keep Uuid first and change it when there are other types share u128.